src/Admin/AdminTagInterface.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Admin;
+
+use Knp\Menu\FactoryInterface;
+use Sonata\AdminBundle\Builder\DatagridBuilderInterface;
+use Sonata\AdminBundle\Builder\FormContractorInterface;
+use Sonata\AdminBundle\Builder\ListBuilderInterface;
+use Sonata\AdminBundle\Builder\RouteBuilderInterface;
+use Sonata\AdminBundle\Builder\ShowBuilderInterface;
+use Sonata\AdminBundle\Exporter\DataSourceInterface;
+use Sonata\AdminBundle\Filter\Persister\FilterPersisterInterface;
+use Sonata\AdminBundle\Model\ModelManagerInterface;
+use Sonata\AdminBundle\Route\RouteGeneratorInterface;
+use Sonata\AdminBundle\Security\Handler\SecurityHandlerInterface;
+use Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * This interface should be implemented to work with the AddDependencyCallsCompilerPass.
+ * All the setter are called by this compiler pass.
+ *
+ * @phpstan-template T of object
+ *
+ * @method void __construct(string $code, string $class, string $controller)
```

Comment:
An interface should not have a `__construct` method

Reply:
I understand, but I don't really know what to do.

Fact is the first and third parameter of the constructor are set by the AddDependencyCallsCompilerPass.
So if you don't have code as the first argument and controller as the third ; it won't work.